From: Mel Gorman <mgorman@suse.de>
Date: Tue, 11 Dec 2012 12:29:31 +0000
Subject: [PATCH] s390/compat: fix compat wrappers for process_vm system calls
 -fix

Patch-mainline: v3.6-rc2
References: fate#314242,bnc#791491

Summary:     kernel: Leverage Cross Memory Attach Functionality for System z
Description: Make the process_vm system calls available on s390.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 arch/s390/kernel/compat_wrapper.S |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/s390/kernel/compat_wrapper.S b/arch/s390/kernel/compat_wrapper.S
index 5274a88..fff4cc4 100644
--- a/arch/s390/kernel/compat_wrapper.S
+++ b/arch/s390/kernel/compat_wrapper.S
@@ -1911,6 +1911,7 @@ sys_setns_wrapper:
 	lgfr	%r3,%r3			# int
 	jg	sys_setns
 
+	.globl compat_sys_process_vm_readv_wrapper
 compat_sys_process_vm_readv_wrapper:
 	lgfr	%r2,%r2			# compat_pid_t
 	llgtr	%r3,%r3			# struct compat_iovec __user *
@@ -1921,6 +1922,7 @@ compat_sys_process_vm_readv_wrapper:
 	stg	%r0,160(%r15)
 	jg	compat_sys_process_vm_readv
 
+	.globl compat_sys_process_vm_writev_wrapper
 compat_sys_process_vm_writev_wrapper:
 	lgfr	%r2,%r2			# compat_pid_t
 	llgtr	%r3,%r3			# struct compat_iovec __user *
